### ISSUE TITLE
Wrong number of entities and overlap between training and inference graph

### ISSUE THREAD

**Issue by bablf** [migalkin/ilpc2022#8]
Hi 

I loaded the ILPC2022Large dataset with pykeen and looked at the dataset. Code is below. 

I was wondering, why the entity counts for the development and test set are wrong. And why there is an intersection between inference and training data. I checked my code several times and can not find an error. 

I would be really thankful if you could tell me what I did wrong or if I overloooked something within the pykeen implementation. 

```
from pykeen.datasets.inductive import ILPC2022Large
dataset = ILPC2022Large()

ilpc_train = pd.DataFrame(dataset.transductive_training.mapped_triples.tolist())
ilpc_ind_inf = pd.DataFrame(dataset.inductive_inference.mapped_triples.tolist())
ilpc_ind_dev = pd.DataFrame(dataset.inductive_validation.mapped_triples.tolist())
ilpc_ind_test = pd.DataFrame(dataset.inductive_testing.mapped_triples.tolist())
print(ilpc_train.shape, ilpc_ind_inf.shape, ilpc_ind_dev.shape, ilpc_ind_test.shape)

ilpc_train_ents = set(ilpc_train[0].tolist() + ilpc_train[2].tolist())
ilpc_ind_inf_ents = set(ilpc_ind_inf[0].tolist() + ilpc_ind_inf[2].tolist())
ilpc_ind_dev_ents = set(ilpc_ind_dev[0].tolist() + ilpc_ind_dev[2].tolist())
ilpc_ind_test_ents = set(ilpc_ind_test[0].tolist() + ilpc_ind_test[2].tolist())
print(len(ilpc_train_ents), len(ilpc_ind_inf_ents), len(ilpc_ind_dev_ents), len(ilpc_ind_test_ents))
print(len(ilpc_train_ents.intersection(ilpc_ind_inf_ents))) # THIS SHOULD BE EMPTY??
```

**Comment by migalkin** [comment]
Hi, 
there is no overlap between training and inference graphs - as you are reading `mapped_triples` that already have numerical entity IDs (not Wikidata QIDs), the [internal pykeen dataloader](https://github.com/pykeen/pykeen/blob/718ce4636fb4b2f88c2a10c0753979c1f7c9a67c/src/pykeen/datasets/inductive/base.py#L232-L244) assigns them with indices in range `[0... num_entities]`. 

In the transductive train of ILPC Large there are 46,626 entities so you might expect to see IDs in range `[0, 46625]` while in the inductive inference we have 29,246 entities, so IDs will be in `[0, 29245]`. 
The entity IDs from the inductive validation and test splits are shared with the **inductive inference graph** so the range of node IDs will also be in `[0, 29245]`.

Obviously, training and inference entities are different Wikidata IDs - you can check the .txt files right here in the repo to make sure. It is useless to use training node IDs (or their vectors) at inference time since those are two disjoint graphs.

**Comment by bablf** [comment]
Ah ok my bad. Thanks for clarifying!
